Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/WebAppModulePresenter.java

```diff
@@ -4,36 +4,19 @@
 
 import com.microsoft.azuretools.core.mvp.model.webapp.AzureWebAppMvpModel;
 import com.microsoft.azuretools.core.mvp.ui.base.MvpPresenter;
-import com.microsoft.tooling.msservices.components.DefaultLoader;
-
-import rx.Observable;
 
 public class WebAppModulePresenter<V extends WebAppModuleView> extends MvpPresenter<V> {
-    private static final String CANNOT_GET_WEB_APPS = "Cannot get web apps.";
     /**
      * Called from view when the view needs refresh.
      */
     public void onModuleRefresh() {
-        Observable.fromCallable(() -> AzureWebAppMvpModel.getInstance().listAllWebApps(true))
-            .subscribeOn(getSchedulerProvider().io())
-            .subscribe(webApps -> DefaultLoader.getIdeHelper().invokeLater(() -> {
-                if (isViewDetached()) {
-                    return;
-                }
-                getMvpView().renderChildren(webApps);
-            }), e -> errorHandler(CANNOT_GET_WEB_APPS, (Exception) e));
+        final WebAppModuleView view = getMvpView();
```

Comment:
The whole block can be written as:
```java
if (!isViewDetached()) {
    getMvpView().renderChildren(AzureWebAppMvpModel.getInstance().listAllWebApps(true));
}
```

This looks simpler to me, but I'm not sticking to it.

Reply:
Will resolve after fixing the block issues.